examples/server-side-rendering/src/app/ssrRouter.ts

```diff
@@ -0,0 +1,63 @@
+import * as qs from 'qs';
+
+export function ssrRouter(readUrl: () => string) {
+  return {
+    read() {
+      const url = readUrl();
+      return qs.parse(url.slice(url.lastIndexOf('?') + 1), {
+        arrayLimit: 99,
+      });
+    },
+    write(routeState) {
+      if (typeof window === 'undefined') return;
+
+      const url = this.createURL(routeState);
+      const title = this.windowTitle && this.windowTitle(routeState);
+
+      if (this.writeTimer) {
+        window.clearTimeout(this.writeTimer);
+      }
+
+      this.writeTimer = window.setTimeout(() => {
+        if (window.location.href !== url) {
+          window.history.pushState(routeState, title || '', url);
+        }
+        this.writeTimer = undefined;
+      }, this.writeDelay);
+    },
+    createURL(routeState) {
+      const url = new URL(readUrl(), 'https://localhost');
```

Comment:
no need to expose baseUrl as the second parameter of `ssrRouter` to replace `'https://localhost'`?